### ISSUE TITLE
Visualizaiton of the learned features

### ISSUE THREAD

**Issue by BIGBALLON** [Vanint/Core-tuning#3]
Hi, thanks for your great work.

I just have a question about the right way to get the visualization results mentioned in the paper:

![image](https://user-images.githubusercontent.com/7837172/158346635-8252798d-4da1-43e4-90ad-b4d2d72784ea.png)

**Comment by YifanZhang07** [comment]
To obtain this visualization, you need an additional 3-dimensional MLP projection between features and classification predictions.  For example, on CIFAR10: 
(1) You need to first train the following network architecture: Feature extractor (512 or 2048 dimensional features) -> 3-dimensional MLP projection (3 dimensional features) -> 10-dimensional classifier (10 dimensional logits). Here, the supervised contrastive loss is enforced on the 3-dimensional feature space. 
(2) After training, you can save the 3-dimensional features of all images for visualization.
(3) Based on the saved features, use your tools (MatLab in our paper) to visualize it.

**Comment by BehzadBozorgtabar** [comment]
Hi, thanks for the answer. Do you also have a ready-to-run script for this experiment and visualization?

**Comment by YifanZhang07** [comment]
> Hi, thanks for the answer. Do you also have a ready-to-run script for this experiment and visualization?

Hi, thanks for your interest. You may refer to: https://github.com/SCUT-AILab/CPGA/tree/main/visualization.
